LearningMissionSimulation/PlaygroundEdvard.cs

```diff
@@ -252,7 +252,42 @@ public void AssignInstructorsToClassrooms()
 
         public void RegisterStudentsForClasses()
         {
-            throw new NotImplementedException();
+            if (ClassroomDictionary.Count != 0)
+            {
+                int maxCount = 0;
+                foreach (Classroom classroom in ClassroomDictionary.Values)
+                {
+                    if (classroom.ItemList.Count < classroom.MaximumCapacity)
+                    {
+                        UpdateStudentList(classroom);
+                        maxCount = classroom.ItemList.Count;
+                    }
+                }
+            }
+        }
+
+        void UpdateStudentList(Classroom classroom)
+        {
+            int studentCount = AttributeGenerator.random.Next(0, (classroom.MaximumCapacity - classroom.ItemList.Count) + 1);
```

Comment:
You should include the active student count in this logic. studentCount can't be bigger than active student count.
The lower limit (for random.Next) can still be 0, if we consider it is a valid simulation case(nobody wants to register for the class). The upper limit should involve the active student count as well.

Reply:
Done.